helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/ClusterModelProvider.java

```diff
@@ -197,6 +371,33 @@ public static ClusterModel generateClusterModelFromCurrentState(
     return toBeAssignedReplicas;
   }
 
+  // <partition, <state, instances set>>
+  private static Map<String, Map<String, Set<String>>> getValidStateInstanceMap(
+      ResourceAssignment assignment, Set<String> activeInstances) {
+    Map<String, Map<String, Set<String>>> stateInstanceMap = getStateInstanceMap(assignment);
+    // Filter to remove all invalid allocations that are not on the active instances.
```

Comment:
Move this up so that it's in the official javadoc for the method?

/*
 * "comment here".
 */?